### PR TITLE
Move all dev dependencies to the root level - Closes #805

### LINKED ISSUE #805: Consider package duplication between root and child packages
### Description
Currently we use `types` folder in root directory to solve `chai` extra types.
However, while converting the packages, I found out it will sometime require type definition on root `package.json`, too. 
Therefore, we should consider making it not duplicate

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
A lot of duplicate devDependencies were in each packages, and it took so long to install, and often failing on some part of npm

### How did I fix it?
Move all the devDependencies to the root level, and add `--hoist` flag to the `bootstrap` command

### How to test it?
```
npm run clean
npm run clean:node_modules
rm -rf node_modules
npm i
npm run build
npm test
```

### Review checklist

* The PR resolves #805
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,13 +42,44 @@
 		"prepush": "npm run lint && npm test"
 	},
 	"devDependencies": {
+		"@types/chai": "4.1.5",
+		"@types/expect": "1.20.3",
+		"@types/jquery": "3.3.9",
+		"@types/mocha": "5.2.5",
+		"@types/node": "10.10.1",
```
Comment (mitsujutsu) [inline package.json]: Also include "types/chai-as-promised":"7.1.0", "types/sinon":"5.0.5", and "types/sinon-chai":"3.2.0"

**Review by mitsuaki-u — CHANGES_REQUESTED** [review]
additional devDependencies in root

File: package.json (hunk 1 above)
Comment (ishantiw) [reply package.json]: Also, we need `"@types/bip39": "2.4.0"` for passphrase

**Review by ishantiw — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (shuse2) [reply package.json]: Are these already exists in `development` branch?

Comment (mitsujutsu) [reply package.json]: Not yet but in near future PRs for typescript. Add now or later?

Comment (shuse2) [reply package.json]: I think it make sense to add them when it requires.
When we the new PR are merged, anyway we need to clean up the `package.json` anyway.

Also, if the another PR goes in first, then we can do opposite

**Review by mitsuaki-u — DISMISSED** [review]

**Pull request closed without merging**